core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -711,6 +716,57 @@ public void setWorkingDirectory(Path path) {
     }
   }
 
+  /**
+   * Check if the input URI is Alluxio on Zookeeper URI.
+   * If it is, set Zookeeper configuration in Hadoop configuration.
+   *
+   * @param uri a uri that may contain Zookeeper information
+   * @param conf hadoop configuration to set
+   */
+  private static void checkAndSetZookeeperConfiguration(URI uri,
+      org.apache.hadoop.conf.Configuration conf) {
+    if (isZookeeperUri(uri)) {
+      conf.setBoolean(PropertyKey.ZOOKEEPER_ENABLED.getName(), true);
```

Comment:
is it safe to modify the passed-in `conf` object? If possible I'd like to avoid risking this.